### DISCUSSION TITLE
mysterious dependency of: babel-preset-react-app@10.0.1

### DISCUSSION THREAD

**Discussion by hsjobeki** [pnpm/pnpm#5732] [Q&A] [answered]
pnpm resolves to depndencies of babel-preset-react-app/10.0.1 
like this:
```yaml
  /babel-preset-react-app/10.0.1:
    resolution: {integrity: sha512-b0D9IZ1WhhCWkrTXyFuIIgqGzSkRIH5D5AmB0bXbzYAB1OBAwHcUeyWW2LorutLWF5btNo/N7r/cIdmvvKJlYg==}
    dependencies:
      '@babel/core': 7.20.2
      '@babel/plugin-proposal-class-properties': 7.18.6_@babel+core@7.20.2
      '@babel/plugin-proposal-decorators': 7.20.2_@babel+core@7.20.2
      '@babel/plugin-proposal-nullish-coalescing-operator': 7.18.6_@babel+core@7.20.2
      '@babel/plugin-proposal-numeric-separator': 7.18.6_@babel+core@7.20.2
      '@babel/plugin-proposal-optional-chaining': 7.18.9_@babel+core@7.20.2
      '@babel/plugin-proposal-private-methods': 7.18.6_@babel+core@7.20.2
      '@babel/plugin-proposal-private-property-in-object': 7.18.6_@babel+core@7.20.2
      '@babel/plugin-transform-flow-strip-types': 7.19.0_@babel+core@7.20.2
      '@babel/plugin-transform-react-display-name': 7.18.6_@babel+core@7.20.2
      '@babel/plugin-transform-runtime': 7.19.6_@babel+core@7.20.2
      '@babel/preset-env': 7.20.2_@babel+core@7.20.2
      '@babel/preset-react': 7.18.6_@babel+core@7.20.2
      '@babel/preset-typescript': 7.18.6_@babel+core@7.20.2
      '@babel/runtime': 7.20.1
      babel-plugin-macros: 3.1.0
      babel-plugin-transform-react-remove-prop-types: 0.4.24
    transitivePeerDependencies:
      - supports-color
    dev: false
```

But the package-json of babel-preset-react-app does not include 

```
@babel/plugin-proposal-private-methods
```

It is not a direct dependency.... because the package.json does not declare it.

however it is a dependency of
```
@babel/preset-env
```
which is as well a dependency here

But i dont understand why it appears here? (at `/babel-preset-react-app/10.0.1`)

I figured out it is required for the `babel-preset-react-app@10.0.1` but i dont understand the mechanism how to figure out that it is a direct dependency and needs to be linked to  `babel-preset-react-app@10.0.1` 

Edit i mean: 

`@babel/plugin-proposal-private-property-in-object`

**Comment by await-ovo** [comment]
![Pasted Graphic 4 copy](https://user-images.githubusercontent.com/41503212/206981417-13771e07-b2b2-4bf5-a435-87c213805d81.png)

**Reply by porsager** [reply]
I think he actually meant `@babel/plugin-proposal-private-property-in-object`.

**Reply by hsjobeki** [reply]
Yes. Sorry messed Up the names.

**Comment by zkochan** [comment, marked as answer]
We are using Yarn's compatibility DB: https://github.com/yarnpkg/berry/blob/f59bbf9f3828865c14b06a3e5cc3ae284a0db78d/packages/yarnpkg-extensions/sources/index.ts#L775-L779

if you want to ignore this, use the [ignore-compatibility-db=true](https://pnpm.io/npmrc#ignore-compatibility-db) setting in `.npmrc`.
